js/gateio.js

```diff
@@ -1847,32 +1847,33 @@ module.exports = class gateio extends Exchange {
         }
         const marketId = this.safeString2 (trade, 'currency_pair', 'contract');
         const symbol = this.safeSymbol (marketId, market);
-        let amountString = this.safeString2 (trade, 'amount', 'size');
-        const priceString = this.safeString (trade, 'price');
-        const costString = Precise.stringAbs (Precise.stringMul (amountString, priceString));
-        const price = this.parseNumber (priceString);
-        const cost = this.parseNumber (costString);
-        const contractSide = Precise.stringLt (amountString, '0') ? 'sell' : 'buy';
-        amountString = Precise.stringAbs (amountString);
-        const amount = this.parseNumber (amountString);
+        let amount = this.safeString2 (trade, 'amount', 'size');
+        const price = this.safeString (trade, 'price');
+        const contractSide = Precise.stringLt (amount, '0') ? 'sell' : 'buy';
+        amount = Precise.stringAbs (amount);
         const side = this.safeString (trade, 'side', contractSide);
         const orderId = this.safeString (trade, 'order_id');
         const gtFee = this.safeString (trade, 'gt_fee');
         let feeCurrency = undefined;
         let feeCost = undefined;
-        if (gtFee === '0') {
-            feeCurrency = this.safeString (trade, 'fee_currency');
-            feeCost = this.safeNumber (trade, 'fee');
-        } else {
-            feeCurrency = 'GT';
-            feeCost = this.parseNumber (gtFee);
+        let fee = undefined;
+        if (gtFee !== undefined) {
+            if (gtFee === '0') {
+                feeCurrency = this.safeString (trade, 'fee_currency');
+                feeCost = this.safeString (trade, 'fee');
+            } else {
```

Comment:
I think the proper way of handling this situation – is to add them both to `fees`, cause the fee may be paid partially in GT. We've seen this with the other exchanges before. Lmk if you can fix for that, if not I'll fix and merge.

Reply:
yeah will fix later tonight

Reply:
@frosty00  What the plan on this PR ?

Reply:
will revise after midterm